lib/ansible/modules/cloud/openstack/os_router.py

```diff
@@ -293,12 +354,34 @@ def _validate_subnets(module, cloud):
 
     if module.params['interfaces']:
         for iface in module.params['interfaces']:
-            subnet = cloud.get_subnet(iface)
-            if not subnet:
-                module.fail_json(msg='subnet %s not found' % iface)
-            internal_subnet_ids.append(subnet['id'])
-
-    return external_subnet_ids, internal_subnet_ids
+            if isinstance(iface, str):
+                subnet = cloud.get_subnet(iface)
+                if not subnet:
+                    module.fail_json(msg='bhujay here subnet %s not found' % iface['subnet'])
```

Comment:
Remove debugging content "bhujay here". Also, iface['subnet'] is not valid when iface is a str.